src/backend/distributed/master/worker_node_manager.c

```diff
@@ -365,8 +381,23 @@ FilterActiveNodeListFunc(LOCKMODE lockMode, bool (*checkFunction)(WorkerNode *))
 List *
 ActivePrimaryNodeList(LOCKMODE lockMode)
 {
+	bool excludeCoordinator = true;
+	EnsureModificationsCanRun();
+	return FilterActiveNodeListFunc(lockMode, WorkerNodeIsPrimary, excludeCoordinator);
```

Comment:
I know it is more code to write, but having both the filter function and `excludeCoordinator ` parameter makes it little unintuitive. How much work is to have different filter functions?

Reply:
My primary reason was that I couldn't find a good name for such filter functions. Now that I think, `NodeIsPrimaryWorker` and `NodeIsPrimary` should work.